Shopify/cli
# [Do not merge] Introduce the `shopify theme update run` command

⚠️ Please, do not merge until https://github.com/Shopify/cli/pull/1909 gets merged.

### WHY are these changes introduced?

Fixes [#420327](https://github.com/Shopify/shopify/issues/420327) part (2/2).

### WHAT is this pull request doing?

This PR introduces the `shopify theme update run` command following the contracts defined by [#420327](https://github.com/Shopify/shopify/issues/420327) and [#420326](https://github.com/Shopify/shopify/issues/420326).

### How to test your changes?

- Run `shopify theme update -h` and notice the new commands appear
- Run `shopify theme update run` and notice it checks your `update_extension.json` script, creates a theme [1], and keeps checking it until the the theme is no longer processing

[1] This step will be revisited, in the future, when we have [#420326](https://github.com/Shopify/shopify/issues/420326) concluded.

![demo_run](https://github.com/Shopify/cli/assets/1079279/d181d410-595f-4a52-8416-2146a4873a07)

### Post-release steps

Update documentation.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
